### PR TITLE
Get First Copy of Gen particle + small fixes

### PR DESCRIPTION
**Pull request by kerstinlovisa**

### CODE REVIEW

File: libs/extensions/include/NanoDimuonVertex.hpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ class NanoDimuonVertex {
   float GetLxyFromPV() { return Lxyz.Perp(); }
 
   float GetCollinearityAngle();
-  float GetPATpTLxyDPhi();
+  float GetMuonpTLxyDPhi(int muonIndex);
```
Comment (jniedzie) [inline libs/extensions/include/NanoDimuonVertex.hpp:52]: I'm not sure what this method returns. Can you make the name more clear?

File: libs/extensions/src/NanoDimuonVertex.cpp [hunk 2]
```diff
@@ -68,14 +68,17 @@ float NanoDimuonVertex::GetCollinearityAngle() {
...
-float NanoDimuonVertex::GetPATpTLxyDPhi() {
-  std::string category = GetVertexCategory();
-  if(category == "PatDSA" && !isDSAMuon1() && isDSAMuon2()) {
-    auto muonFourVector = asNanoMuon(muon1)->GetFourVector();
-    TVector3 ptVector(muonFourVector.Px(), muonFourVector.Py(), muonFourVector.Pz());
-    return ptVector.DeltaPhi(Lxyz);
+float NanoDimuonVertex::GetMuonpTLxyDPhi(int muonIndex) {
+  std::shared_ptr<PhysicsObject> muon;
+  if (muonIndex == 1) muon = muon1;
+  else if (muonIndex == 2) muon = muon2;
```
Comment (jniedzie) [inline libs/extensions/src/NanoDimuonVertex.cpp:74]: Why do we count from 1 and not from 0?

File: libs/extensions/src/NanoDimuonVertex.cpp [hunk 3]
```diff
@@ -68,14 +68,17 @@ float NanoDimuonVertex::GetCollinearityAngle() {
...
+  else {
+    warn() << "Invalid muon index " << muonIndex << " in NanoDimuonVertex::GetMuonpTLxyDPhi" << endl;
+    return -5;
```
Comment (jniedzie) [inline libs/extensions/src/NanoDimuonVertex.cpp:77]: Why -5?

File: libs/extensions/include/NanoDimuonVertex.hpp (hunk 1 above)
Comment (kerstinlovisa) [reply libs/extensions/include/NanoDimuonVertex.hpp:52]: it's delta phi between the muon pT and Lxy, I don't know what else to call it. Any suggestions are welcomed :)

File: libs/extensions/src/NanoDimuonVertex.cpp (hunk 2 above)
Comment (kerstinlovisa) [reply libs/extensions/src/NanoDimuonVertex.cpp:74]: Actually I've always named the muons in the vertex fit as muon1 and muon2, so here I thought it confusing to give input 1 and get return value for muon 2

File: libs/extensions/src/NanoDimuonVertex.cpp (hunk 3 above)
Comment (kerstinlovisa) [reply libs/extensions/src/NanoDimuonVertex.cpp:77]: Because -5 is outside of [-pi, pi]

Comment (jniedzie) [reply libs/extensions/src/NanoDimuonVertex.cpp:77]: Oh, I see! Ok.

File: libs/extensions/include/NanoDimuonVertex.hpp (hunk 1 above)
Comment (jniedzie) [reply libs/extensions/include/NanoDimuonVertex.hpp:52]: Right, now I get it. Idk if maybe `GetDphiBetweenMuonPtAndLxy(...)` could work?

File: libs/extensions/src/NanoDimuonVertex.cpp (hunk 2 above)
Comment (jniedzie) [reply libs/extensions/src/NanoDimuonVertex.cpp:74]: Alright.

File: libs/extensions/include/NanoDimuonVertex.hpp (hunk 1 above)
Comment (kerstinlovisa) [reply libs/extensions/include/NanoDimuonVertex.hpp]: Change it now

Comment (kerstinlovisa) [reply libs/extensions/include/NanoDimuonVertex.hpp]: I will also push to the other ttalps PR as well

**Review by jniedzie — APPROVED** [review]

**Pull request merged**